mlflow/pyfunc/__init__.py

```diff
@@ -3027,9 +3028,10 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
         )
     elif python_model is not None:
         if callable(python_model):
-            # TODO: support input_example and TypeFromExample for callables
+            saved_example = _save_example(mlflow_model, input_example, path, example_no_conversion)
             # first argument is the model input
             type_hints = _extract_type_hints(python_model, input_arg_index=0)
+            type_hint_from_example = _is_type_hint_from_example(type_hints.input)
             if not _signature_cannot_be_inferred_from_type_hint(type_hints.input):
```

Comment:
This will be True if use TypteFromExample.
Can we refactor a bit to simplify the logic between callable & PythonModel?

Reply:
Sure, I think there are many variables whose scope is a bit large (e.g. saved_example, signature_from_type_hints) and the logic flow is not easy to follow. Let me unify the logic of callable and Pythonmodel in this PR, but let's revisit and split the script into multiple smaller chunks.